eq-author/src/components/CommentsPanel/index.js

```diff
@@ -0,0 +1,202 @@
+import CustomPropTypes from "custom-prop-types";
+
+import COMMENT_QUERY from "./commentsQuery.graphql";
+import COMMENT_ADD from "./createNewComment.graphql";
+
+import styled from "styled-components";
+import ScrollPane from "components/ScrollPane";
+import { colors } from "constants/theme";
+
+import TextArea from "components/Forms/TextArea";
+import { Field, Label } from "components/Forms";
+import Button from "components/buttons/Button";
+import ButtonGroup from "components/buttons/ButtonGroup";
+
+const CommentsPane = styled.div`
+  background: ${colors.white};
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: 100%;
+  padding: 0;
+  margin: 0;
+  font-size: 1em;
+`;
+
+const CommentSection = styled.div`
+  padding: 0.7em 1em;
+  border-bottom: 1px solid ${colors.lightMediumGrey};
+`;
+
+const ShowComment = styled.div`
+  align-items: left;
+  padding: 0.5em 1em;
+  margin-bottom: 5px;
+  border: 1px solid ${colors.lightGrey};
+  background-color: ${colors.lighterGrey};
+`;
+
+const CommentHeaderContainer = styled(Field)`
+  display: flex;
+  justify-content: flex-start;
+  -webkit-align-items: start;
+  padding: 0;
+  margin: 0;
+  margin-bottom: 8px;
+`;
+
+const AvatarWrapper = styled.div`
+  flex-grow: 0;
+  margin-right: 8px;
+`;
+
+const AvatarOuter = styled.div`
+  width: 36px;
+  height: 36px;
+  border-radius: 50%;
+  background-color: ${colors.primary};
+  text-align: center;
+`;
+
+const AvatarInner = styled.div`
+  color: ${colors.white};
+  position: relative;
+  display: inline-block;
+  width: 100%;
+  height: 0;
+  padding: 25% 0;
+`;
+
+const NameWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  -webkit-flex-direction: column;
+  justify-content: space-between;
+  align-items: start;
+  flex-grow: 2;
+`;
+
+const FlexLabel = styled.div`
+  font-size: 1.1em;
+  align-items: center;
+  height: 22px;
+`;
+
+const DateField = styled("span")`
+  font-weight: normal;
+  font-size: 0.8em;
+  color: ${colors.grey};
+`;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const CommentsPanel = ({
+  match: {
+    params: { pageId },
+  },
+}) => {
+  const [comment, setComment] = useState("");
+  const { loading, error, data } = useQuery(COMMENT_QUERY, {
+    variables: {
+      input: { pageId },
+    },
+  });
+
+  const [createComment] = useMutation(COMMENT_ADD);
+
+  const handleChange = event => {
+    setComment(event.target.value);
+  };
+
+  const handleSubmit = event => {
+    if (comment) {
+      createComment({
+        variables: {
+          input: {
+            pageId,
+            commentText: comment,
+          },
+        },
+      });
+      setComment("");
+    }
+    event.preventDefault();
+  };
+
+  const getInitials = name => {
+    const initials = name.replace(/[^a-zA-Z- ]/g, "").match(/\b\w/g);
+    return initials.join("").toUpperCase();
+  };
+
+  //make sure qraphQL has returned before rendering
```

Comment:
I don't think this comment needs to be here, we do this quite a lot throughout the application and what the code does is fairly self explanatory.